scripts/create_version.sh

```diff
@@ -0,0 +1,167 @@
+#!/bin/bash
+
+# Copyright Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+parse_input() {
+    [[ $1 =~ ^release-([0-9])\.([0-9]+)\.([0-9]+)$ ]]
+
+    MAJOR="${BASH_REMATCH[1]}"
+    MINOR="${BASH_REMATCH[2]}"
+    PATCH="${BASH_REMATCH[3]}"
+
+    if [ "${MAJOR}" == '' ]; then
+        echo "Target format error: should be 'release-x.x.x', got '$1'"
+        exit 1
+    fi
+
+    echo "Creating release for ${MAJOR}.${MINOR}.${PATCH}..."
+
+    # patch release
+    if [ "${PATCH}" != '0' ]; then
+        scripts/patch_release.sh "${MAJOR}" "${MINOR}" "${PATCH}"
+        exit 0
+    fi
+
+    # major/minor release
+    CURR_MINOR="${MAJOR}.${MINOR}"     # current version
+    PREV_MINOR="${MAJOR}.$((MINOR-1))" # previous version
+    NEXT_MINOR="${MAJOR}.$((MINOR+1))" # next version
+
+    if [ "${DRY_RUN}" == '1' ]; then
+        CURR_MINOR="${CURR_MINOR}-dry-run"
+        git checkout "${MASTER}"
+        git pull "${ISTIOIO_GIT_SOURCE}" "${MASTER}"
+    fi
+
+    # for a major release x.0, find the latest minor release
+    if [ "${MINOR}" == '0' ]; then
+        LAST_MINOR_OF_PREV_MAJOR=$(
+            git branch -a |
+            grep "release-$((MAJOR-1))." |
+            sed -r "s/^.*release-$((MAJOR-1))\.([0-9]+)$/\1/" |
+            sort -n |
+            tail -1
+        )
+        PREV_MINOR="$((MAJOR-1)).${LAST_MINOR_OF_PREV_MAJOR}"
+    fi
+
+    echo "Previous minor release: ${PREV_MINOR}"
+    echo "Upcoming minor release: ${NEXT_MINOR}"
+}
+
+archive_old_release() {
+    echo -e "\nStep 1: archive the old release branch"
+    if [ "${DRY_RUN}" == '1' ]; then
+        echo "Skipping step 1 in dry run"
```

Comment:
This is probably something we should still do in the dry run, right? We should just be pushing the archived site to a different branch. We just shouldn't check in the changes to the archived branch

Reply:
That would mean we'd create yet another branch like `master-dry-run` off master? Or are you suggesting only pushing a single folder `archive/v1.x` to a separate branch?

Reply:
I mean that we should still go back to the old branch, archive it, and add it as a folder in the release branch